### ISSUE TITLE
Got high loss after restoring parameters from ckpt

### ISSUE THREAD

**Issue by xiaoxTM** [tensorflow/models#2952]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**: research
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: customized for task of segmentation
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 14.04 (64bit)
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.20
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: CUDA-8 CUDNN-6
- **GPU model and memory**: Tesla-P100
- **Exact command to reproduce**: 
```bash
TRAIN_DIR=object_detection/ckpt/ssd_inception
PIPELINE_CONFIG_PATH=object_detection/samples/configs/ssd_inception_v2_coco.config

python3 object_detection/train.py \
            --logtostderr \
            --pipeline_config_path=${PIPELINE_CONFIG_PATH} \
            --train_dir=${TRAIN_DIR}
```

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Hi everyone here:
    I have got a issue when training (fine-tuning) Tensorflow Object Detection API model.
I restored parameters from fine-tuning-ckpt (saved models to another directory, e.g., TRAIN_DIR), and after fine-tuning with big learning rate, the loss decreased as I expected. As it went lower and lower (e.g., 2.0), it is time to use smaller learning rate. So I moved the saved parameters from TRAIN_DIR to fine-tuning-ckpt, and removed all files under TRAIN_DIR. When I re-run the program, I expected the loss to be low (about 2.0), but got 300+. I tried several times and got the same results every time. So I guess the reason is that, the program I was running did not use the saved parameters (or did not restore from TRAIN_DIR), so I printed the variables_to_restore but found that all variable names. Does anybody have any idea or have the same problem ?

Best Regards

**Comment by idorozenberg** [comment]
Also happened to me. Please help with this issue. Makes it impossible to work with the SSD meta-model if I can't use transfer learning properly.

**Comment by idorozenberg** [comment]
@drpngx @derekjchow. Are you looking into this?

Thank you.

**Comment by sarah-zhu** [comment]
I also had the same issue when training SSD model.

**Comment by idorozenberg** [comment]
Tried Adam, regular momentum.

**Comment by sarah-zhu** [comment]
It seems the variables are restored from pre-trained model, but the program is not using the restored variables.
Please help!

**Comment by dextroza** [comment]
Also happend to me. I took pretrained model ssd_mobilenet_v2_coco and tried to continue to train on COCO dataset. First few losses are 300+. I expected less than 10.

**Comment by JulienSiems** [comment]
Same here with a Mask RCNN inception V2 model pretrained on a custom data set.

**Comment by dcarnino** [comment]
+1 with Faster RCNN inception resnet v2 atrous coco on a custom data set.

**Comment by tT0NG** [comment]
+1 with SSD inception v2 coco on custom data set.

**Comment by TitusTom** [comment]
same as dcarnino. Happened on multiple occasions. 
+2 with Faster RCNN inception resnet v2 on a custom dataset

**Comment by DSRYhh** [comment]
+1 with SphereFace.
I use Dataset API and Adam optimizer with default parameters.

**Comment by wt-huang** [comment]
Closing as this is resolved
